File: 0024-Swap-Nodes-in-Pairs/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(0,head)
        prev = dummy

        while(prev.next is not None and prev.next.next is not None):
            a = prev.next
            b = a.next
            upc = b.next

            prev.next = b
            b.next = a
            a.next= upc
            prev = a

        
        return dummy.next

File: 0024-Swap-Nodes-in-Pairs/solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* swapPairs(struct ListNode* head) {
    struct ListNode* dummy = malloc(sizeof(struct ListNode));
    dummy->next= head;
    struct ListNode* prev = dummy;


    while (prev->next != NULL && prev->next->next != NULL) {
        struct ListNode* a = prev->next;
        struct ListNode* b = a->next;
        struct ListNode* next = b->next;

        prev->next = b;
        b->next = a;
        a->next = next;
        prev = a;
    }

    return dummy->next;

}
